# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: tools/profiler/public/GeckoProfiler.h
# Commit: a01db300cf22
# Full Hash: a01db300cf221298a1937d7ba6c82d6466526353
# Author: Gerald Squelart <gsquelart@mozilla.com>
# Date: 2020-10-17 09:14:56
# Regressor Bug: 1670954
# File Overlap Count: 5
# Description:
#   Bug 1670954 - profiler_main_thread_id() and profiler_is_main_thread() - r=gregtatum
#   
#   These functions will be useful to get the main thread id, or check if we're in it, in some public code (e.g., markers).
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D93735
# ==============================================================================

diff -r c0a3393e4a80 -r a01db300cf22 tools/profiler/public/GeckoProfiler.h
--- a/tools/profiler/public/GeckoProfiler.h	Fri Oct 16 21:06:13 2020 +0000
+++ b/tools/profiler/public/GeckoProfiler.h	Fri Oct 16 22:06:53 2020 +0000
@@ -640,6 +640,14 @@
 // Get the current thread's ID.
 int profiler_current_thread_id();
 
+extern const int scProfilerMainThreadId;
+
+inline int profiler_main_thread_id() { return scProfilerMainThreadId; }
+
+inline bool profiler_is_main_thread() {
+  return profiler_current_thread_id() == profiler_main_thread_id();
+}
+
 // An object of this class is passed to profiler_suspend_and_sample_thread().
 // For each stack frame, one of the Collect methods will be called.
 class ProfilerStackCollector {